nixos/modules/services/web-apps/funkwhale/funkwhale.nix

```diff
@@ -0,0 +1,329 @@
+{config, lib, pkgs, ...}:
+
+with lib;
+
+let
+  pythonEnv = (
+    let pythonRequirements = import ./requirements.nix { inherit pkgs; };
+    in
+      with pkgs; 
+      python36.withPackages(ps: with ps; (builtins.attrValues pythonRequirements.packages))
+    );
+  cfg              = config.services.funkwhale;
+  funkwhaleHome    = config.users.extraUsers.funkwhale.home;
+  funkwhaleEnvFile = (import ./funkwhale.env.nix)       { pkgs = pkgs; cfg = cfg; };
+  funkwhaleEnv = {
+    FUNKWHALE_ENV_FILE = "${funkwhaleEnvFile}";
+  };
+in 
+{ 
+
+  options = {
+    services.funkwhale = {
+      enable = mkEnableOption "funkwhale";
+
+      api_ip = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = ''
+          Funkwhale API IP
+        '';
+      };
+
+      api_port = mkOption {
+        type = types.str;
+        default = "5000";
+        description = ''
+          Funkwhale API Port
+        '';
+      };
+
+      hostname = mkOption {
+        type = types.str;
+        default = "yourdomain";
+        description = ''
+          The definitive, public domain you will use for your instance
+        '';
+      };
+
+      protocol = mkOption {
+        type = types.str;
+        default = "https";
+        description = ''
+          http or https
+        '';
+      };
+
+      email_config = mkOption {
+        type = types.str;
+        default = "consolemail://";
+        description = ''
+          Configure email sending using this variale
+          By default, funkwhale will output emails sent to stdout
+          here are a few examples for this setting
+          consolemail://         # output emails to console (the default)
```

Comment:
Enclose this in either a `<literal>` (keyword) or `<programlisting>` (multiline) xml section. Build the docs yourself to find out whether they render correctly.

Reply:
thanks. fixed